src/GuiLib/Classes/Slider.lua

```diff
@@ -206,7 +211,9 @@ function init(self)
 		spring:Update(dt)
 		local x = spring.x
 		if (x ~= last) then
-			local scalePos = (bPos + (x * bSize) - frame.AbsolutePosition[axis]) / frame.AbsoluteSize[axis]
+			local aPos = frame.AbsolutePosition[axis] -- AbsolutePosition and AbsoluteSize can return wrong values on the first frame 
```

Comment:
My only issue w/ this fix is that it's basically voodoo. It does work, but the reasoning as to why it works is most likely hidden under the hood of the Roblox engine.

The commit(s) I made force the slider bounds to update on the first render frame available. This fixes the issue and I think is more explicit in what it's trying to achieve b/c it doesn't rely on the voodoo 😆 .

Reply:
That's very fair, this definitely is a more explicit way to communicate what should be happening.
I think it's a little yucky having a check in the RenderStepped loop that only ever runs once, wouldn't calling updateBounds before making the RenderStepped connection accomplish the same thing?
Other than that, looks good, thanks!

Reply:
Previously I did call it before the renderstep connection:
https://github.com/EgoMoose/Rbx-Gui-Library/pull/2/commits/dab019eca623a57d2b70d3fe4f67a77081c01574#diff-e22b8313f35b992aba31c8248654778c7a168360f510d28baf465ecc6849b531L151

I've experienced issues somewhat similar to this before like textbounds not updating the renderframe you set text. Idk if that's fixed now, but the old solution was to defer a frame.